app/routes/goal_routes.py

```diff
@@ -1 +1,119 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..models.goal import Goal
+from .route_utilities import validate_model, model_from_request
+from ..models.task import Task
+from ..db import db
+
+bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+
+@bp.post("")
+def create_goal():
+    new_goal = model_from_request(cls=Goal, request=request)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+
+@bp.get("")
+def get_all_goals():
+    title_param = request.args.get("title")
+    sort = request.args.get("sort")
+
+    query = db.select(Goal)
+
+    if title_param:
+        query = query.where(goal.title.ilike(f"%{title_param}%"))
+
+    goals = db.session.scalars(query)
+
+    goals_response = []
+
+    for goal in goals:
+        goals_response.append(goal.to_dict())
+
+    return goals_response
+
+
+@bp.get("/<goal_id>")
+def get_single_goal(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+
+    return {
+        "goal": goal.to_dict()
+    }
+
+
+@bp.get("/<goal_id>/tasks")
+def get_tasks_for_goal(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+
+    query = db.select(Task).where(Task.goal_id == goal_id)
+
+    goal_dict = goal.to_dict()
+
+    tasks = db.session.scalars(query)
+
+    goal_dict["tasks"] = [task.to_dict() for task in tasks]
```

Comment:
Instead of creating a Select object, we could just get the `tasks` from `goal`.
```suggestion
    goal_dict = goal.to_dict()
    goal_dict["tasks"] = [task.to_dict() for task in goal.tasks]
```